src/content/reference/device-os/firmware.md

```diff
@@ -12979,6 +13005,1754 @@ int8_t measurePower() const;
+
+
+### isWritable() [Ledger class]
+
+Returns true if the ledger can be written to. Cloud to device ledgers cannot be written to.
+
+```cpp
+// PROTOTYPE
+bool isWritable() const;
+```
+
+### onSync(OnSyncCallback) [Ledger class]
+
+Registers a function to be called when the ledger is synchronized. This method takes a C function and an `arg` value. The arg can be a pointer to a C++ object instance, a pointer to some other structure, or can be 0 if you don't need an additional argument.
+
+```cpp
+// PROTOTYPE
+int onSync(OnSyncCallback callback, void* arg = nullptr);
+
+// CALLBACK DEFINITION
+typedef void (*OnSyncCallback)(Ledger ledger, void* arg);
+
+// CALLBACK PROTOTYPE
+void myCallback(Ledger ledger, void* arg)
+```
+
+### onSync(OnSyncFunction) [Ledger class]
+
+Registers a function to be called when the ledger is synchronized. This method takes a std::function which can be a C++ lambda, which can be useful for calling a method implemented in a non-static C++ class.
+
+```cpp
+// PROTOTYPE
+int onSync(OnSyncFunction callback);
+
+// CALLBACK DEFINITIONS
+typedef std::function<void(Ledger)> OnSyncFunction;
+
+// CALLBACK PROTOTYPE
+void myCallback(Ledger ledger)
+
+// CALLBACK LAMBDA
+myLedger.onSync([](Ledger ledger) {
+  // Code to run when synchronization completes goes here
+}
+```
+
+### remove() [Ledger class]
+
+Remove any data associated with a ledger from the device.
+
+The device must not be connected to the Cloud. The operation will fail if the ledger with the given name is in use.
+
+The data is not guaranteed to be removed in an irrecoverable way.
+
+See also [Ledger::removeAll](#removeall-ledger-class-) to remove all ledger data and includes a code example.
+
+```cpp
+// PROTOTYPE
+static int remove(const char* name);
+```
+
+### removeAll() [Ledger class]
+
+Remove any ledger data from the device.
+
+The device must not be connected to the Cloud. The operation will fail if the ledger with the given name is in use.
+
+The data is not guaranteed to be removed in an irrecoverable way.
+
+```cpp
+// PROTOTYPE
+static int removeAll();
+
+// EXAMPLE
+#include "Particle.h"
+
+SYSTEM_MODE(SEMI_AUTOMATIC);
+
+SYSTEM_THREAD(ENABLED);
+
+SerialLogHandler logHandler(LOG_LEVEL_INFO);
+
+void setup() {
+    // Remove any ledger data from the device.    
+    // The device must not be connected to the Cloud. The operation will fail if any of the ledgers are in use.
+    Ledger::removeAll();
+
+    Particle.connect();
+}
+
+void loop() {
+}
+
+```
+
+## LedgerData
+
+The `LedgerData` class provides a subset of methods of the `Variant` class that are relevant to map operations. The Ledger class [set()](#set-ledger-class-) and [get()](#get-ledger-class-) take a LedgerData object. 
+
+Because a `Variant` is transparently converted into `LedgerData` you may never need to create a `LedgerData` object. See [Ledger sensor](/getting-started/logic-ledger/ledger-sensor/) for an example of this technique.
```

Comment:
This is not true for a conversion in the opposite direction though. I'm wondering if we should allow it too.